### PR TITLE
Use new result API

### LINKED ISSUE #5849: [Tech Debt] Remove usage of deprecated `startActivityForResult` & `onActivityResult` APIs  in favour of decoupled result APIs
Remove usage of deprecated `startActivityForResult` & `onActivityResult` APIs  in favour of decoupled result APIs. 

> While the underlying [startActivityForResult()](https://developer.android.com/reference/android/app/Activity#startActivityForResult(android.content.Intent,%20int)) and [onActivityResult()](https://developer.android.com/reference/android/app/Activity#onActivityResult(int,%20int,%20android.content.Intent)) APIs are available on the Activity class on all API levels, Google strongly recommends using the Activity Result APIs introduced in AndroidX [Activity](https://developer.android.com/jetpack/androidx/releases/activity) and [Fragment](https://developer.android.com/jetpack/androidx/releases/fragment) classes.

For more on what benefits it offers refer here https://developer.android.com/training/basics/intents/result#custom

P.S: We already have newer code using this API. Only some place that needs refactor.

### PR DESCRIPTION
**Pull request by parneet-guraya**
Fixes #5849

What changes did you make and why?

- Migrate Activities that are launched with expectancy of getting result back, to new result api.

#### For ease of review, these are the  functionalities that are affected and have been tested. 

- File picking -> Camera, Gallery, Custom; from every place one or more of these options can be used have been refactored.
- getting result back from `ImageFragment` launched from `CustomSelectorActivity` 
- getting result of  voice input.
-  result from image editing from `EditActivity`

**Tests performed (required)**
`betaDebug` & `prodDebug` tested on Oneplust 9RT 5G , Android 14

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java [hunk 1]
```diff
@@ -148,20 +152,21 @@ private void showLocationOffDialog(Activity activity, int dialogTextResource,
                 () -> {
                     Toast.makeText(activity, activity.getString(toastTextResource),
                         Toast.LENGTH_LONG).show();
-                    initiateCameraUpload(activity);
+                    initiateCameraUpload(activity,resultLauncher);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java:155]: space after `,`

**Review by nicolas-raoul — COMMENTED** [review]
I have tested the features you mentioned, they seem to be working great on Pixel 9 too. :-)
I will try some uploads tomorrow.

Meanwhile, I noticed a few places missing a comma after `,` would you mind fixing them?

Thanks a lot!

**Comment by parneet-guraya** [comment]
Also what do we use for updating branches? I saw you used merge, but I was using rebase until now. Do we follow any standard?

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java [hunk 2]
```diff
@@ -204,18 +210,18 @@ private void askUserToAllowLocationAccess(Activity activity,
     /**
      * Initiate gallery picker
      */
-    public void initiateGalleryPick(final Activity activity, final boolean allowMultipleUploads) {
-        initiateGalleryUpload(activity, allowMultipleUploads);
+    public void initiateGalleryPick(final Activity activity,ActivityResultLauncher<Intent> resultLauncher ,final boolean allowMultipleUploads) {
+        initiateGalleryUpload(activity,resultLauncher ,allowMultipleUploads);
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java:214]: space after `,` not before

**Comment by parneet-guraya** [comment]
Added requested changes https://github.com/commons-app/apps-android-commons/pull/5875/commits/b68256d7f0d6d28ca20622e2df194ceaf2791301

**Review by nicolas-raoul — APPROVED** [review]
Great work, thanks!

**Pull request merged**